sample_consensus/include/pcl/sample_consensus/impl/sac_model_cone.hpp

```diff
@@ -217,33 +219,35 @@ pcl::SampleConsensusModelCone<PointT, PointNT>::selectWithinDistance (
   for (std::size_t i = 0; i < indices_->size (); ++i)
   {
     Eigen::Vector4f pt ((*input_)[(*indices_)[i]].x, (*input_)[(*indices_)[i]].y, (*input_)[(*indices_)[i]].z, 0.0f);
-    Eigen::Vector4f n  ((*normals_)[(*indices_)[i]].normal[0], (*normals_)[(*indices_)[i]].normal[1], (*normals_)[(*indices_)[i]].normal[2], 0.0f);
 
     // Calculate the point's projection on the cone axis
     float k = (pt.dot (axis_dir) - apexdotdir) * dirdotdir;
     Eigen::Vector4f pt_proj = apex + k * axis_dir;
 
-    // Calculate the direction of the point from center
-    Eigen::Vector4f pp_pt_dir = pt - pt_proj;
-    pp_pt_dir.normalize ();
-
     // Calculate the actual radius of the cone at the level of the projected point
     Eigen::Vector4f height = apex - pt_proj;
     double actual_cone_radius = tan(opening_angle) * height.norm ();
-    height.normalize ();
-
-    // Calculate the cones perfect normals
-    Eigen::Vector4f cone_normal = sinf (opening_angle) * height + std::cos (opening_angle) * pp_pt_dir;
 
     // Approximate the distance from the point to the cone as the difference between
     // dist(point,cone_axis) and actual cone radius
-    double d_euclid = std::abs (pointToAxisDistance (pt, model_coefficients) - actual_cone_radius);
+    const double d_euclid = (1.0 - normal_distance_weight_) * std::abs (pointToAxisDistance (pt, model_coefficients) - actual_cone_radius);
+    if (d_euclid > threshold) // Early termination: cannot be an inlier
+      continue;
+
+    // Calculate the direction of the point from center
+    Eigen::Vector4f pp_pt_dir = pt - pt_proj;
+    pp_pt_dir.normalize ();
+
+    // Calculate the cones perfect normals
+    height.normalize ();
+    Eigen::Vector4f cone_normal = sinf (opening_angle) * height + std::cos (opening_angle) * pp_pt_dir;
 
     // Calculate the angular distance between the point normal and the (dir=pt_proj->pt) vector
+    Eigen::Vector4f n  ((*normals_)[(*indices_)[i]].normal[0], (*normals_)[(*indices_)[i]].normal[1], (*normals_)[(*indices_)[i]].normal[2], 0.0f);
     double d_normal = std::abs (getAngle3D (n, cone_normal));
     d_normal = (std::min) (d_normal, M_PI - d_normal);
 
-    double distance = std::abs (normal_distance_weight_ * d_normal + (1.0 - normal_distance_weight_) * d_euclid);
+    double distance = std::abs (normal_distance_weight_ * d_normal + d_euclid);
```

Comment:
I agree, but for the early termination condition it is better (faster) like this. I could
1. do the same for `d_normal` as I did for `d_euclid` to have it consistent for both
2. rename the variables to make their meaning clearer
3. and/or compute a modified threshold `threshold/(1.0 - normal_distance_weight_)` to use in the early termination condition

Reply:
In this instance, which termination condition are you referring to?

Reply:
[This one (line 234)](https://github.com/PointCloudLibrary/pcl/pull/4424/files#diff-31d009f62f193ad924b18f1c2d51202eR234).

Reply:
I'd recommend at least pt. 2 and advice against pt. 3 (mult is better than division). Doing pt. 1 would bring symmetry, but it's not a necessity since something like `normal_distance_weight * d_normal + weighted_euclid_distance` has enough content to warn the reader.

Reply:
Regarding pt. 3: that modified threshold only has to be computed once, before the for-loop, since `threshold` and `normal_distance_weight_` are the same for all points

Reply:
I renamed the variables (pt. 2).